src/ast/mod.rs

```diff
@@ -3314,6 +3314,8 @@ pub enum Statement {
     CreateRole {
         names: Vec<ObjectName>,
         if_not_exists: bool,
+        /// Whether ROLE or USER keyword was used
+        keyword: RoleKeyword,
```

Comment:
Could we use something like this representation instead? I think its a somewhat more invasive changes but clearer API wise.
```rust
struct CreateUserOrRole {
  names: Vec<ObjectName>,
  if_not_exists: bool,
  // Postgres
  login: Option<bool>,
  inherit: Option<bool>,
  // ...
}

Statement::CreateUser(CreateUserOrRole)
Statement::CreateRole(CreateUserOrRole)

struct AlterUserOrRole {
  name: Ident
}

Statement::AlterUser(AlterUserOrRole)
Statement::AlterRole(AlterUserOrRole)
```

Reply:
Ah I'm not sure I followed, do we mean snowflake has CreateUser supported by the parser already?